certificate-manager/src/main/java/io/strimzi/certs/IpAndDnsValidation.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class to help with validation of IP addresses and DNS addresses
+ */
+public class IpAndDnsValidation {
+    // Pattern for matching IPv4 address (requires additional check for segments to be equal or less than 255 - this is done in the isValidIpv4Address method)
+    private static final Pattern IPV4_ADDRESS = Pattern.compile("\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}");
+
+    // Pattern for uncompressed IPv6 address -> case-insensitive, tolerates leading 0s
+    private static final Pattern IPV6_ADDRESS = Pattern.compile("^(?:[\\da-fA-F]{1,4}:){7}[\\da-fA-F]{1,4}$");
+
+    // Pattern for compressed IPv6 address -> case-insensitive, tolerates leading 0s
+    private static final Pattern IPV6_COMPRESSED_ADDRESS = Pattern.compile("^((?:[\\dA-Fa-f]{1,4}(?::[\\dA-Fa-f]{1,4})*)?)::((?:[\\dA-Fa-f]{1,4}(?::[\\dA-Fa-f]{1,4})*)?)$");
+
+    // Pattern for DNS names
+    private static final Pattern DNS_NAME = Pattern.compile("^(" +
+            // a single char dns name
+            "[a-zA-Z\\d]|" +
+            // can't begin or end with -                followed by more labels of same
+            "[a-zA-Z\\d][a-zA-Z\\d\\-]{0,61}[a-zA-Z\\d])(\\.([a-zA-Z\\d]|[a-zA-Z\\d][a-zA-Z\\d\\-]{0,61}[a-zA-Z\\d]))*$");
+
+    // Pattern used to normalize the segments of the IPv6 address
+    private static final Pattern IPV6_LEADING_ZEROS_TRIMMING = Pattern.compile("(^|:)0+([\\da-f]+)(:|$)");
+
+    /**
+     * Returns true if the name is a valid DNS name. That means that it matches the DNS_NAME pattern
```

Comment:
Add a link the the RFC that defines the grammar for DNS names, perhaps?

Reply:
I'm a bit reluctant to do this. The IPv4 validation or the DNS validation are not new. They were just moved to not have them across 3 different classes. I addressed the other comments you had for them. But I do not really know what relation does this have to the RFC and how well it covers it.